src/content/styleguide.mdx

```diff
@@ -380,6 +379,31 @@ const Header: AppFunctionComponent = () => {
 export default Header
 ```
 
+### Props of styled components
+
+If props definition of a component is short and used individually, add it inline. It will be easier to read, because usually there are a lot of styled components defined in a one file.
+
+```typescript
+const Button = styled.button<{ pending: boolean }>`...`
+```
+
+If a styled component becomes to be more advanced one (with many props or its props are reuseable), wonder about creating a dedicated interface for it.
+
+```typescript
+// Could be hard to read
+const Button = styled.button<{ pending: boolean, wide: boolean, empty: boolean, ... }>`...`
```

Comment:
I think prettier in this case will format it so each prop would be in its own separate line

Reply:
After prettier formatting it will be still the same case IMO 🤔 
```
const Button = styled.button<{
   pending: boolean,
   wide: boolean,
   empty: boolean,
   ...
}>`...`
```

Reply:
If we want to be `consistent` I think same rule would have to be applied to other functionalities from SC, like `css helper` or `attrs`.

Reply:
I would say, in general, styled components are still components. Since we always put "normal" component's props in the dedicated interface, we should do the same with SCs. 🤔

Reply:
Styled components are still components, but use cases are different  - that's why there is different (less restricted) rule proposed.